vahidkay-meta/facebook-for-woocommerce
# Fixes the item not found error by using filter in the product endpoint

### Changes proposed in this Pull Request: 

Updates the ProductCatalog/Products/Id API call to use endpoint filter instead of directly getting the product with id.

This API call is mainly used for checking if the product item exists on Meta. When the item does not exists it threw an error. With this change the endpoint will stop erroring and will return empty list as response.


### Detailed test instructions:

1. Create any product on admin page that will be synced to Meta.
2. Check for errors.

### Changelog entry

> Dev - Use product GET endpoint filter function to reduce the item not found errors.
